python/tank/bootstrap/manager.py

```diff
@@ -921,39 +922,78 @@ def _start_engine(self, tk, engine_name, entity, progress_callback=None):
+                # It's possible that a tk-core is being used that didn't come from
+                # the app_store. This might be the case where a site config has been
+                # locked off, and populated with a tk-core cloned from Github. In that
+                # case, it'll have "HEAD" as its version. In that situation, we don't
+                # actually know if this is a "new" core, or something super old. Given
+                # that, we need to try going the legacy route here just to see if we
+                # might actually have an old core.
+                if is_version_head(tk.version) and is_shotgun_engine:
+                    try:
+                        engine = self._legacy_start_shotgun_engine(tk, engine_name, entity, ctx)
+                    except Exception:
+                        # We'll want to raise the original exception here. We don't
+                        # really know whether this is a legacy core or not, so if both
+                        # the legacy and new code paths failed, we'll raise the exception
+                        # from the new code path.
+                        log.debug(
+                            "Attempted legacy engine start path for Shotgun engine, which failed. "
+                            "This attempt was made because the tk-core version is HEAD, which means "
+                            "we don't know if it's new or old. As such, when the new-style bootstrap "
+                            "failed, the legacy pathway was attempted."
+                        )
+                        raise exc
+                else:
+                    # In this case, we know we're in a new enough core, but that we
+                    # just failed for some legitimate reason.
+                    raise exc
 
         log.debug("Launched engine %r" % engine)
 
         self._report_progress(progress_callback, self._BOOTSTRAP_COMPLETED, "Engine launched.")
 
         return engine
 
+    def _legacy_start_shotgun_engine(self, tk, engine_name, entity, ctx):
+        """
+        Starts the tk-shotgun engine by way of the legacy "start_shotgun_engine"
+        method provided by tank.platform.engine.
+
+        :param tk: Bootstrapped :class:`~sgtk.Sgtk` instance.
+        :param engine_name: Name of the engine to start up.
+        :param entity: Shotgun entity used to resolve a project context.
+        :type entity: Dictionary with keys ``type`` and ``id``, or ``None`` for the site.
+        """
+        # bootstrapping into a shotgun engine with an older core
+        # we perform this special check to make sure that we correctly pick up
+        # the shotgun_xxx.yml environment files, even for older cores.
+        # new cores handles all this inside the tank.platform.start_shotgun_engine
+        # business logic.
+        log.debug(
+            "Target core version is %s. Starting shotgun engine via legacy pathway." % tk.version
+        )
```

Comment:
No on this one. It'll end up bubbling up to the browser since that info log will end up in stdout by way of the cache_commands and execute_commands subprocesses, which themselves bootstrap.